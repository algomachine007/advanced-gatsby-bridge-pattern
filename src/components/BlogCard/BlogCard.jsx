import React from 'react';
import PropTypes from 'prop-types';

import { Link } from 'gatsby';
import ContentfulImage from 'utility-components/ContentfulImage/ContentfulImage';

import svgPropTypesShape from 'utility-components/propTypeShapes/svgPropTypesShape';
import { formatDate } from 'helpers/formatDate';
import truncateText from 'helpers/truncateText';
import styles from './BlogCard.module.scss';

const BlogCard = ({ title, mainImage, date, page }) => (
  <Link className={styles.blogCard} to={`/${page.path}`}>
    <div className={styles.image}>
      <ContentfulImage image={mainImage} className={styles.imageHeight} />
    </div>

    <div className={styles.content}>
      <time>{formatDate(date)}</time>
      <h2>{truncateText(title, 60)}</h2>
    </div>
  </Link>
);

export default BlogCard;

BlogCard.defaultProps = {
  page: {
    path: '',
  },
  title: '',
  mainImage: {},
  date: '',
};

BlogCard.propTypes = {
  title: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  mainImage: svgPropTypesShape.isRequired,
  page: PropTypes.shape({
    path: PropTypes.string.isRequired,
  }),
};
