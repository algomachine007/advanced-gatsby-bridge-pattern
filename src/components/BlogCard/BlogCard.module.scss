.blogCard {
  display: flex;
  position: relative;
  min-height: 300px;
  transition: all 0.3s ease-in-out;
  min-width: 100%;

  &:hover {
    transform: translate(0, -10px);
    cursor: pointer;
    .content {
      background-color: $color-primary-dark;
      opacity: 1;
    }
  }

  @media (min-width: $breakpoint-tablet--sm) {
    min-width: 100%;
  }

  @media (min-width: $breakpoint-tablet) {
    min-width: 45%;
  }

  @media (min-width: $breakpoint-desktop) {
    min-width: 580px;
    min-height: 585px;
  }

  .content {
    position: absolute;
    bottom: 0;
    height: calc(100% / 2.2);
    min-width: 100%;
    background-color: $color-primary-dark;
    padding: 20px;
    color: $color-standard-white;
    opacity: 0.8;

    @media (min-width: $breakpoint-desktop) {
      height: calc(100% / 3.5);
      padding: 40px 24px;
    }

    time {
      font-size: 12px;
      font-weight: $font--bold;
      letter-spacing: 1px;
      margin-bottom: 8px;
    }

    h2 {
      font-size: $font--20;
      font-weight: $font--bold;
      letter-spacing: -1px;
      line-height: $line-height--wiggle;

      @media (min-width: $breakpoint-desktop) {
        font-size: $font--30;
        line-height: $line-height--wiggle;
      }
    }
  }
}

.image {
  width: 100%;

  @media (min-width: $breakpoint-tablet) {
    object-fit: cover;
  }
}

.imageHeight {
  height: 100%;
}
