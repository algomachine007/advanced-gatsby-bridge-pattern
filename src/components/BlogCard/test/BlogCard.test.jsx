import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import BlogCard from '../BlogCard';


describe('BlogCard', () => {
  const mockPage = {
    path: '/page',
  };

  it('renders correctly with appropriate props', () => {
    const card = shallow(
      <BlogCard
        mainImage="image.jpg"
        date="Aug 1, 2022"
        title="B2B payments a reality with Real-Time..."
        page={mockPage}
      />,
    );

    expect(toJson(card)).toMatchSnapshot();
  });
});
