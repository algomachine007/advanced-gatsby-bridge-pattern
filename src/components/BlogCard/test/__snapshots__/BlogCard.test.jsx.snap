// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogCard renders correctly with appropriate props 1`] = `
<mockConstructor
  className="blogCard"
  to="//page"
>
  <div
    className="image"
  >
    <ContentfulImage
      className="imageHeight"
      image="image.jpg"
      productScreenVideoPlayButtonImage={null}
      productScreenYouTubeVideoId={null}
    />
  </div>
  <div
    className="content"
  >
    <time>
      Aug 1, 2022
    </time>
    <h2>
      B2B payments a reality with Real-Time...
    </h2>
  </div>
</mockConstructor>
`;
